src/legacy/core_plugins/kibana/public/visualize/np_ready/listing/visualize_listing.js

```diff
@@ -124,4 +124,10 @@ export function VisualizeListingController($injector, createNewVis) {
   this.listingLimit = uiSettings.get('savedObjects:listingLimit');
 
   addHelpMenuToAppChrome(chrome, docLinks);
+
+  $scope.$on('$destroy', () => {
```

Comment:
This is added for cases when we navigate away without interacting with dialog. E.g. back button in browser which navigates us to other app.